### PR TITLE
Logging and Key copying improvements to Worker

### PR DESCRIPTION
**Pull request by reset**
* We now create both an stdout & stderr log for builds
* The worker will copy the correct origin keys based on the origin of the plan it is building
* Secret origin key is now automatically downloaded if not present on worker before building
* Added an optional error field to the job message which includes a NetError if the job failed

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -1085,6 +1105,16 @@ pub fn router(depot: Depot) -> Result<Chain> {
         origin_secret_key_create: post "/origins/:origin/secret_keys/:revision" => {
             XHandler::new(upload_origin_secret_key).before(basic.clone())
         },
+        origin_secret_key_latest: get "/origins/:origin/secret_keys/latest" => {
+            if depot.config.insecure {
+                XHandler::new(download_latest_origin_secret_key)
+            } else {
+                // JW TODO: Maybe we shouldn't allow administrators to access this endpoint. This
+                // was created for workers to pull secret keys from the API before performing work.
+                // We should probably create a new feature flag for "WORKER" or something similar.
```
Comment (raskchanky) [inline components/builder-depot/src/server.rs]: Great idea.

Comment (reset) [inline components/builder-depot/src/server.rs]: I took care of this and removed the comment ;)

File: components/builder-worker/src/runner/logger.rs [hunk 2]
```diff
@@ -13,26 +13,60 @@
 // limitations under the License.
 
 use std::fs::File;
-use std::io::Write;
+use std::io::{BufRead, BufReader, Write};
+use std::process;
 
 use super::workspace::Workspace;
 
+const EOL_MARKER: &'static str = "\n";
```
Comment (raskchanky) [inline components/builder-worker/src/runner/logger.rs]: Is this going to cause problems on Windows?  Once upon a time, Windows used "\r\n" as the line terminator.  Is that still a thing?  My windows knowledge is admittedly antiquated.

**Comment by raskchanky** [comment]
1 minor comment but nothing to prevent merging.

gif-keyboard-633404910066708473

File: components/builder-worker/src/runner/logger.rs (hunk 2 above)
Comment (reset) [inline components/builder-worker/src/runner/logger.rs]: We control the output of the log so this shouldn't become a problem on Windows - we're not going to use their line endings anyway

**Comment by reset** [comment]
gif-keyboard-8217805143126967702

Thanks for the review!

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/181470903).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/181470903).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**